q8/main.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+/* global output */
+const SEARCH_WORD = 'foo'
+const REPLACE_WORD = 'uryyyy!!'
+
+// const hash = { text: 'foo' }
+// const hash = {
+//   main: {
+//     first: { text: 'foobar' },
+//     second: { text: 'fizzbuzzfoo', child: { text: 'foobar' } },
+//   },
+// }
+const MODEL_HASH = {
+  main: {
+    first: { text: 'foobar' },
+    second: { text: 'fizzbuzz', child: { text: 'foobar' } },
+  },
+  sub: {
+    first: { text: 'fizzbuzz', child: { text: 'foobar' } },
+    second: {
+      third: { text: 'barfoo', child: { text: 'foobar' } },
+      forth: { child: { text: 'jit_foo_foo' } },
+    },
+  },
+  text: 'foofava',
+  foo: '外れ',
+}
+try {
+  main()
+} catch (error) {
+  output(error.message)
+}
+function main() {
+  const objectToString = function(obj) {
+    return JSON.stringify(obj, null, '  ')
+  }
+  output('hash', objectToString(MODEL_HASH))
+  output('toUry', objectToString(toUry(MODEL_HASH)))
+}
+
+/**
+ * text が保持する文字列内のfooをuryyyy!!に変換する
+ * @param {Object} hash 変換対象
+ */
+function toUry(hash) {
+  if (!hash || typeof hash !== 'object') return hash
+  for (const key of Object.keys(hash)) {
+    if (hash['text']) {
+      hash['text'] = replaceWord(hash['text'])
```

Comment:
オブジェクトのプロパティはこのようにしてもアクセス可能です。
```suggestion
      hash.text = replaceWord(hash.text)
```

プロパティ名にハイフンなどが入っていたりすると使えませんが、通常はこの方式でOKです。 👍

Reply:
こちらのほうが、直感的というか、properyアクセスというか
読みやすいですね。

a7a7209